### ISSUE TITLE
Update Contrib libraries to v0.14.0

### ISSUE THREAD

**Issue by JordanMartinez** [purescript-contrib/governance#35]
When we decide to start updating these, we'll use this issue to track which repos we've done and which we have yet to do, similar to purescript/purescript#3942

I'm merely opening this issue for the time being and expect to edit this opening comment with more details later.

---

Tracking issues and pull requests to merge before 0.14:
https://github.com/issues?q=is%3Aopen+archived%3Afalse+user%3Apurescript-contrib+label%3A%22fix+before+0.14%22

**Comment by JordanMartinez** [comment]
One thing my script isn't doing is checking for usages of `Proxy`. I'm gong to update my scripts to recursively search through the src and test directory to find such usages.

**Comment by JordanMartinez** [comment]
[See the original comment](https://github.com/purescript-contrib/purescript-profunctor-lenses/pull/122#issuecomment-738190901) for source of content below:
<hr>
Role annotations are only necessary to relax the nominal roles of foreign data types or to restrict the roles of phantom parameters when they should satisfy some implicit invariant.

There’s no such types here: `Forget` and `Tagged` have phantom parameters but as far as I know there’s no invariant to preserve by restricting their roles.

I have audited the purescript-contrib org again to confirm but all the types benefiting from more precise roles have pull requests already:

- [x] https://github.com/purescript-contrib/purescript-ace/pull/32
- [x] https://github.com/purescript-contrib/purescript-aff-bus/pull/20
- [x] https://github.com/purescript-contrib/purescript-aff/pull/194
- [x] https://github.com/purescript-contrib/purescript-avar/pull/20
- [x] https://github.com/purescript-contrib/purescript-css/pull/120
- [x] https://github.com/purescript-contrib/purescript-nullable/pull/29
- [x] https://github.com/purescript-contrib/purescript-options/pull/30
- [x] https://github.com/purescript-contrib/purescript-react/pull/175

**Comment by kl0tl** [comment]
The following packages mention `SProxy` (see https://github.com/search?q=SProxy+user%3Apurescript-contrib+language%3APureScript&type=code): `argonaut-codecs`, `argonaut-generic`, `profunctor-lenses`, `pathy`, `react` and `uri`.

Only `argonaut-codecs` mention `RLProxy` (see https://github.com/search?q=RLProxy+user%3Apurescript-contrib+language%3APureScript&type=code)

`RProxy`, `BProxy` and `OProxy` aren’t mentioned in any contrib library.

**Comment by JordanMartinez** [comment]
Just FYI. Some of the contrib libraries aren't showing up in my list because their repos were transferred to this organization. Thus, the 'grep' solution I'm using to find all contrib libraries won't work because the URL to the repo refers to the original repo (e.g. `slamdata/purescript-pathy`), not the current repo (e.g. `purescript-contrib/purescript-pathy`).

I think this is also why `purescript-graphs` didn't show up in my list for core libraries.

**Comment by JordanMartinez** [comment]
This one still hasn't gotten a review: https://github.com/purescript-contrib/purescript-argonaut-codecs/pull/90

**Comment by JordanMartinez** [comment]
These repos don't show up in my list for contrib. Checkbox below means I've created a PR for them. If it's unchcked. Something is blocking it, or I haven't submitted a PR for it yet.
- [x] affjax - blocked by form-urlencoded and web-xhr
- [x] aff-bus
- [x] css
- [x] precise - blocked by quickcheck-laws
- [x] routing
- [x] formatters - blocked by parsing
- [x] fork
- [x] concurrent-queues
- [x] uri - blocked by parsing and js-uri
- [x] string-parsers

**Comment by JordanMartinez** [comment]
Besides the repos above that weren't on my list, the only other repos that haven't had a PR submitted are:
- [x] react-dom - blocked by "react", "web-dom","web-events"
- [x] github-actions-toolkit - blocked by "aff-promise","node-buffer","node-path","node-streams"
- [x] ace - blocked by "web-dom","web-events","web-file","web-html","web-storage","web-uievents"

**Comment by thomashoneyman** [comment]
Oh, shoot, the `purescript-web` and `purescript-node` organizations have to be done too, that's right.

**Comment by JordanMartinez** [comment]
Yeah.... haha

**Comment by JordanMartinez** [comment]
Here's a situation where MonadZero's `guard` is used: https://github.com/purescript-contrib/purescript-formatters/blob/main/test/src/DateTime.purs#L108

**Comment by JordanMartinez** [comment]
I've submitted all PRs to all contrib repos, but not all of these will build due to dependencies on node and web libraries.

Also, the `jquery` lib I mentioned above is owned by `paf31`, so I don't think it should be updated here.

**Comment by JordanMartinez** [comment]
I added all the libraries that were updated to the `prepare-0.14` package set's `updatedLibs.dhall` file to verify that everything still compiles. I added `purescript-graphs` to the package set and added `purescript-concurrent-queues` to the package set.

During one of the changes I made, CI complained about [pathy and aff-bus](https://github.com/purescript/package-sets/runs/1504344596), which surprised me because these have been updated. When I looked for their files, I found that those two packages were defined in the `slamdata.dhall` file ([aff-bus](https://github.com/purescript/package-sets/blob/prepare-0.14/src/groups/slamdata.dhall#L27-L32) and [pathy](https://github.com/purescript/package-sets/blob/prepare-0.14/src/groups/slamdata.dhall#L154-L170)). Should these packages (and others that were transferred to the `contrib` org be moved into the `purescript-contrib.dhall` file? We've already done something like that for the `numbers` repo that was transferred to the purescript org.

**Comment by thomashoneyman** [comment]
Yea, let's transfer those!

**Comment by thomashoneyman** [comment]
Quick heads up here: the `-web` and `-node` organizations are on their way to be updated for 0.14, which will leave us just merging outstanding issues and pull requests across organization libraries for the rest of the month. I'll also be working with @garyb on a new release for Halogen and the core team will be making sure Pursuit is ready to go for 0.14. With all of this in motion we're getting quite close to the finish line!

**Comment by thomashoneyman** [comment]
At this point we only have these libraries to go, which are blocked by the missing encode / decode URI component functions -- solved if we do https://github.com/purescript-contrib/purescript-form-urlencoded/pull/20#issuecomment-739418901:
- ~https://github.com/purescript-contrib/purescript-affjax/pull/158~
- ~https://github.com/purescript-contrib/purescript-uri/pull/57~
- ~https://github.com/purescript-contrib/purescript-routing/pull/76~
- ~https://github.com/purescript-contrib/purescript-form-urlencoded/pull/20~

~We also have https://github.com/purescript-contrib/purescript-css.~

**Comment by JordanMartinez** [comment]
CSS was merged I believe.

**Comment by thomashoneyman** [comment]
The four libraries which are blocked (above) can be resolved once this library is published:

https://github.com/purescript-contrib/purescript-js-uri

**Comment by thomashoneyman** [comment]
These libraries are now all complete! I am going to make a followup issue about issues marked 'fix for 0.14', but the libraries themselves are now all compatible with 0.14.
